saleor/discount/models.py

```diff
@@ -454,6 +454,9 @@ class BaseDiscount(models.Model):
         choices=DiscountType.CHOICES,
         default=DiscountType.MANUAL,
     )
+    # `unique_type` is a field to ensure, that only single specific discount type
+    # can be associated with discounted object
+    unique_type = models.CharField(max_length=64, null=True, blank=True)
```

Comment:
Maybe worth to use `choices` here?

Reply:
Initially I used `DiscountType` here. But setting this char field gives us more freedom to handle uniqueness of discounts without changing the API or creating a migration. Does it make sense?

Reply:
You are right, we should use `DiscountType` here